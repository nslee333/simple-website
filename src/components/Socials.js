import slack_logo from "../../public/images/slack_logo.png";
import meetup_logo from "../../public/images/meetup_logo.png";
import youtube_logo from "../../public/images/youtube_logo.svg.png";
import Image from "next/image";

export default function Socials() {
  return (
    <>
      <div className="my-10">
        <div className="m-5 flex justify-center ">
          <h5 className="text-white xs:text-2xl md:text-3xl">
            Check us out on...
          </h5>
        </div>
        <div className="flex justify-center">
          <div className="sm:w-[50rem]">
            <div className="flex xs:grid xs:w-60 xs:justify-center sm:flex sm:w-auto sm:justify-around md:flex">
              <div className="xs:ml-[4.2rem] sm:my-0 sm:ml-0">
                <div className="sm:mt-[1rem]">
                  <a href="https://www.meetup.com/BendJS/" target="_blank" className="flex">
                    <Image
                      src={meetup_logo}
                      className="xs:w-[80px]"
                      alt="Meetup Logo"
                    />
                    <div className="pt-6">
                      <h5 className="text-2xl text-white xs:text-lg sm:text-xl">
                        /BendJS
                      </h5>
                    </div>
                  </a>
                </div>
              </div>
              <div className="flex justify-center xs:my-12 sm:my-8 sm:mr-10">
                <a href="https://bendjs.slack.com/ssb/redirect" target="_blank">
                  <Image
                    src={slack_logo}
                    className="aspect-auto xs:w-[175px] sm:w-[175px]"
                    alt="Slack Logo"
                  />
                </a>
              </div>
              <div className="flex justify-center sm:mr-10 sm:mt-[.59em]">
                <a href="https://www.youtube.com/channel/UCCT744K1sGX-lPtaEivXCDQ" target="_blank">
                  <Image
                    src={youtube_logo}
                    className="xs:h-[50px] xs:w-[70px] sm:my-[1.22rem]"
                    alt="Youtube Logo"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </>
  );
}
