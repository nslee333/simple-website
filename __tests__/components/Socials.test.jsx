import { render, screen } from "@testing-library/react";
import Socials from "../../src/components/Socials";
import { toHaveAttribute } from "@testing-library/jest-dom";
import renderer from "react-test-renderer";

test("Socials renders without crashing", () => {
  render(<Socials />);
});

test("Socials renders correctly according to snapshot.", () => {
  const tree = renderer.create(<Socials />).toJSON();

  expect(tree).toMatchSnapshot();
});

test("Meetup logo has link to Meetup page.", () => {
  render(<Socials />);

  expect(
    screen.getByRole("link", { name: "Meetup Logo /BendJS" })
  ).toHaveAttribute("href", "https://www.meetup.com/BendJS/");
});

test("Slack logo has link to Slack page.", () => {
  render(<Socials />);

  expect(screen.getByRole("link", { name: "Slack Logo" })).toHaveAttribute(
    "href",
    "https://bendjs.slack.com/ssb/redirect"
  );
});

test("Youtube logo has link to Youtube page.", () => {
  render(<Socials />);

  expect(screen.getByRole("link", { name: "Youtube Logo" })).toHaveAttribute(
    "href",
    "https://www.youtube.com/channel/UCCT744K1sGX-lPtaEivXCDQ"
  );
});
