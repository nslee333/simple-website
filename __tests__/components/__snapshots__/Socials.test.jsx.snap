// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Socials renders correctly according to snapshot. 1`] = `
<div
  className="my-10"
>
  <div
    className="m-5 flex justify-center "
  >
    <h5
      className="text-white xs:text-2xl md:text-3xl"
    >
      Check us out on...
    </h5>
  </div>
  <div
    className="flex justify-center"
  >
    <div
      className="sm:w-[50rem]"
    >
      <div
        className="flex xs:grid xs:w-60 xs:justify-center sm:flex sm:w-auto sm:justify-around md:flex"
      >
        <div
          className="xs:ml-[4.2rem] sm:my-0 sm:ml-0"
        >
          <div
            className="sm:mt-[1rem]"
          >
            <a
              className="flex"
              href="https://www.meetup.com/BendJS/"
              target="_blank"
            >
              <img
                alt="Meetup Logo"
                className="xs:w-[80px]"
                data-nimg="1"
                decoding="async"
                height={40}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={40}
              />
              <div
                className="pt-6"
              >
                <h5
                  className="text-2xl text-white xs:text-lg sm:text-xl"
                >
                  /BendJS
                </h5>
              </div>
            </a>
          </div>
        </div>
        <div
          className="flex justify-center xs:my-12 sm:my-8 sm:mr-10"
        >
          <a
            href="https://bendjs.slack.com/ssb/redirect"
            target="_blank"
          >
            <img
              alt="Slack Logo"
              className="aspect-auto xs:w-[175px] sm:w-[175px]"
              data-nimg="1"
              decoding="async"
              height={40}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={40}
            />
          </a>
        </div>
        <div
          className="flex justify-center sm:mr-10 sm:mt-[.59em]"
        >
          <a
            href="https://www.youtube.com/channel/UCCT744K1sGX-lPtaEivXCDQ"
            target="_blank"
          >
            <img
              alt="Youtube Logo"
              className="xs:h-[50px] xs:w-[70px] sm:my-[1.22rem]"
              data-nimg="1"
              decoding="async"
              height={40}
              loading="lazy"
              onError={[Function]}
              onLoad={[Function]}
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style={
                {
                  "color": "transparent",
                }
              }
              width={40}
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
